### PR TITLE
Adding custom cache function

### PR DESCRIPTION
**Pull request by erez-work**

### CODE REVIEW

File: src/cache.js [hunk 1]
```diff
@@ -3,13 +3,20 @@ import hash from "object-hash";
...
+    cache.get,
+    cache.set,
+    f
+  )
+};
+
+export const withCacheAsyncCustom = (get, set, f) => async (...args) => {
+  const cacheKey = hash.sha1(args);
+  const value = await get(cacheKey)
+  if (value){
```
Comment (nitzo) [inline src/cache.js]: I would explicitly check if this is null / undefined (what ever you think) as `0` might be a valid response right ? 
and then fix cache.get to transform this into undefined

File: src/cache.test.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const { withCacheAsync } = require("./cache");
+
+test("async cache", async () => {
+  const foo = withCacheAsync((x) => x*x)
+  expect(await foo(5)).toEqual(25);
```
Comment (nitzo) [inline src/cache.test.js]: this is not really testing the cache right ? maybe better to return the timestamp and then call it twice to see it doesn't change

**Review by nitzo — APPROVED** [review]

File: src/cache.js (hunk 1 above)
Comment (erez-work) [reply src/cache.js]: changed to ramda's isNil

File: src/cache.test.js (hunk 2 above)
Comment (erez-work) [reply src/cache.test.js]: Right, using Math.random now

**Pull request merged**